doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,299 @@
+  Produces a stall (Category LdHz) and shouldn’t forward data.
+
+State Specific Behaviour
+""""""""""""""""""""""""
+Some instructions will behave differently depending upon the state of the processor (e.g. the privilege level the instruction executes at, CSR settings or whether the processor is in debug mode).
+
+* Instruction illegal in U Mode (cover execution in U and M mode).
+
+  * Mret
+  * Wfi
+  * Access to M-mode CSR.
+
+* Debug mode only instructions (cover execution in and out of debug mode).
+
+  * Dret
+  * Access to debug CSRs.
+
+    * ``dcsr``
+    * ``dpc``
+    * ``dscratch0``
+    * ``dscratch1``
+
+  * Access to trigger CSRs (also possible in M mode, cover execution in M mode, debug mode and U mode).
+
+    * ``tselect``
+    * ``tdata1``
+    * ``tdata2``
+    * ``tdata3``
+    * ``mcontext``
+    * ``scontext``
+
+* Loads/stores with ``mstatus.mprv`` set and unset.
+* EBreak behaviour in U/M mode with different ``dcsr.ebreakm`` / ``dcsr.ebreaku`` settings.
+
+Pipeline State
+^^^^^^^^^^^^^^
+Each pipeline stage has some associated state.
+
+* IF stage full or empty
+* WB stage full or empty
+* ID stage full or empty -  effectively covered by instruction categories (Category None meaning the stage is empty).
+* Controller (within ID stage) state machine states
+
+  * Possible transitions between these states.
+    Those marked with a '*' are of particular interest and should be crossed with instruction categories and other coverpoints as appropriate to fully explore the transitions.
+
+    * ``RESET`` -> ``BOOT_SET``
+    * ``BOOT_SET`` -> ``FIRST_FETCH``
+    * ``FIRST_FETCH`` -> ``DECODE``
+    * ``FIRST_FETCH`` -> ``IRQ_TAKEN``
+    * ``FIRST_FETCH`` -> ``DBG_TAKEN_IF``
+    * ``DECODE`` -> ``FLUSH`` *
+    * ``DECODE`` -> ``DBG_TAKEN_IF`` *
+    * ``DECODE`` -> ``IRQ_TAKEN`` *
+    * ``IRQ_TAKEN`` -> ``DECODE``
+    * ``DBG_TAKEN_IF`` -> ``DECODE``
+    * ``DBG_TAKEN_ID`` -> ``DECODE``
+    * ``FLUSH`` -> ``DECODE`` *
+    * ``FLUSH`` -> ``DBG_TAKEN_ID``
+    * ``FLUSH`` -> ``WAIT_SLEEP``
+    * ``FLUSH`` -> ``IRQ_TAKEN`` *
+    * ``FLUSH`` -> ``DBG_TAKEN_IF`` *
+    * ``WAIT_SLEEP`` -> ``SLEEP``
+    * ``SLEEP`` -> ``FIRST_FETCH``
+
+Exceptions/Interrupts/Debug
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Exceptions, interrupts and debug entry can all cause control flow changes combined with CSR writes and privilege level changes and work quite similarly within the controller but not identically.
+Furthermore they can all occur together and must be appropriately prioritised (consider a instruction with hardware trigger point matching it, that causes some exception and an interrupt is raised the cycle it enters the ID/EX stage)
+
+* Exception from instruction fetch error (covered by FetchError instruction category).
+* Exception from instruction PMP violation.
+* Exception from illegal instruction (covered by Illegal instruction categories).
```

Comment:
Nit: Maybe "covered by **the** FetchError..." and then "covered by the various illegal instruction categories"? I don't think it makes sense to capitalise "Illegal", because that's not the name of a category.